nvflare/private/fed/server/job_runner.py

```diff
@@ -161,6 +163,32 @@ def _stop_run(self, run_number, fl_ctx: FLContext):
             if err:
                 self.log_error(fl_ctx, f"Failed to abort the server for run_.{run_number}")
 
+    def _delete_run(self, run_number, sites: dict, fl_ctx: FLContext):
+        """Delete the run workspace
+
+        Args:
+            run_number: run_number
+            sites: participating sites
+            fl_ctx: FLContext
+
+        Returns:
+
+        """
+        engine = fl_ctx.get_engine()
+
+        client_sites = list(sites.keys())
+        admin_server = engine.server.admin_server
+        message = Message(topic=TrainingTopic.DELETE_RUN, body="")
+        message.set_header(RequestHeader.RUN_NUM, str(run_number))
+        self.log_debug(fl_ctx, f"Send delete_run command to the site for run:{run_number}")
+        replies = self._send_to_clients(admin_server, client_sites, engine, message)
+        if not replies:
+            self.log_error(fl_ctx, f"Failed to send delete_run command to clients for run_{run_number}")
+
+        err = engine.delete_run_number(run_number)
+        if err:
+            self.log_error(fl_ctx, f"Failed to delete_run the server for run_.{run_number}")
```

Comment:
Suggest these error messages should be change to

```f"Failed to XXX for run folder: {run_number}```

Reply:
Or if there are other places using this "run_" pattern we should also change them.